frontend/src/actions/story.js

```diff
@@ -9,25 +9,54 @@ import {
   SUCCESS,
   FETCH_STORY_START,
   FETCH_STORY,
+  INVALID_STORY,
+  ERROR,
 } from "./types";
+import { fields } from "../routes/CriticalQuestions/fields";
+
+const mandatoryFields = [fields.CURRENT_LOCATION];
```

Comment:
In case you're bored enough, [here's](https://github.com/pyritewolf/cosmotecton/blob/master/frontend/src/components/form/Form/index.js) an example on how I usually solve the whole validation dilemma (and it's consumed [here](https://github.com/pyritewolf/cosmotecton/blob/master/frontend/src/views/LogIn/index.js)).

It's not the only solution, and the path you took is pretty much as valid as any other - like I told you, it's one of React's recurring problems, but I thought you might find it interesting 😄